api_v2/src/notifications/notifications.repository.ts

```diff
@@ -41,7 +44,48 @@ export class NotificationRepository {
     return updatedRequest as ESLChangeShiftRequest;
   }
 
-  async updateNotificationType({
+  async getEslOffsetRequestIdFromNotification(notificationId: number): Promise<number> {
+    const notification = await this.prisma.notification.findUnique({
+      where: { id: notificationId },
+      select: { relatedEntityId: true },
+    });
+
+    if (!notification || !notification.relatedEntityId) {
+      throw new Error('Notification not found or does not have a valid related ESL Offset Request ID.');
+    }
+
+    return notification.relatedEntityId;
+  }
+
+  async updateEslOffsetApproval(requestId: number, isApproved: boolean): Promise<ESLOffset> {
+    const updatedOffset = await this.prisma.eslOffset.update({
+      where: { id: requestId },
+      data: { isLeaderApproved: isApproved },
+    });
+
+    if (!updatedOffset) {
+      throw new Error('Failed to update ESL Offset request.');
+    }
+
+    return updatedOffset as ESLOffset;
+  }
+
+  async updateEslOffsetNotificationType(
+    notificationId: number,
+    teamLeaderId: number,
+    isApproved: boolean,
+  ): Promise<void> {
+    await this.prisma.notification.update({
```

Comment:
same as here, Prisma .update() will throws an error if the record isn't found . You can wrap that in try/catch 

```suggestion
try {

  async updateEslOffsetNotificationType(

  await this.prisma.notification.update({ your function here});
} catch (error) {
  throw new Error('Failed to update notification type for ESL Offset.');
}
```

Reply:
@Mjdelima20 Noted I will refactor